Stochastic duplication of VoxelIntersection data, on GitHub Actions

I can't reproduce this locally, but this error started appearing during the TTY work:

```
======================================================================
FAIL: test_single_voxel (test_connectivity.TestVoxelIntersection)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/bsb-core/bsb-core/tests/test_connectivity.py", line 562, in test_single_voxel
    self.assertEqual(2, len(pre_locs), "expected 2 connections")
AssertionError: 2 != 4 : expected 2 connections

----------------------------------------------------------------------
```

Similarly, sometimes the `test_cells_placed` places 39 instead of 40 cells.

--- a/bsb/services/pool.py
+++ b/bsb/services/pool.py
@@ -412,7 +412,7 @@ class Job(abc.ABC):
                 self._enqueue(self._pool)
 
     def _enqueue(self, pool):
-        if not self._deps and self._status is not JobStatus.CANCELLED:
+        if not self._deps and self._status is JobStatus.PENDING:
             # Go ahead and submit ourselves to the pool, no dependencies to wait for
             # The dispatcher is run on the remote worker and unpacks the data required
             # to execute the job contents.

--- a/tests/test_connectivity.py
+++ b/tests/test_connectivity.py
@@ -457,7 +457,6 @@ class TestConnWithSubCellLabels(
             )
 
 
-@unittest.skip("https://github.com/dbbs-lab/bsb-core/issues/820")
 class TestVoxelIntersection(
     RandomStorageFixture,
     NetworkFixture,

